highfidelity/hifi
# Added method to test namespace to save JSON object to a file.

# Description
A method has been added to the Test namespace, available when running from the command line with the `--testScript` parameter.  This method saves an object to a file, in JSON format.  A new parameter `testResultsLocation` defines where the file will be saved.

The new command is Test.saveObject(<_object name_>, <_file name_>).

For example:
```
        var results = {
            elapsedTime: 0.05
        }
        
        Test.saveObject(results, "results.txt");

```

# Testing
The test will run a performance test in the dev-avatarislanfd domain.
1.  Create a folder to store the results in.  Assume it is C:\testing.
2.  Run the following line in a consoles:
```
interface.exe --testScript https://github.com/highfidelity/hifi_tests/blob/master/tests/performance/testAuto.js?raw=true quitWhenFinished --testResultsLocation C:\testing
```

3.  Verify that a file named results.txt has been created in C:\testing.  It's contents should be similar to the following:
```
{
    "elapsedTime": 11.544
}
```
